core/src/main/java/org/apache/calcite/sql/dialect/OracleSqlDialect.java

```diff
@@ -54,10 +54,20 @@
           case VARCHAR:
             // Maximum size of 4000 bytes for varchar2.
             return 4000;
+          case CHAR:
+            return 2000;
           default:
             return super.getMaxPrecision(typeName);
           }
         }
+        @Override public int getDefaultPrecision(SqlTypeName typeName) {
+          switch (typeName) {
+          case VARCHAR:
```

Comment:
I noticed the following in the official description of oracle:
Since Oracle Database 12c, you can specify the maximum size of 32767 for the VARCHAR2 data type. Oracle uses the MAX_STRING_SIZE parameter for controlling the maximum size. If the MAX_STRING_SIZE is STANDARD , then the maximum size for VARCHAR2 is 4000 bytes.
https://www.oracletutorial.com/oracle-basics/oracle-varchar2/
I suggest adding a database version or description here?

Reply:
Thanks a lot, I add the comment for this kind of situation.